File: Clases_Instancias/2/estudiantes_lista.py
class Estudiante:
    def __init__(self, nombre, edad, promedio):
        self.nombre = nombre
        self.edad = edad
        self.promedio = promedio

    def mostrarInfo(self):
        print(f"Nombre: {self.nombre}")
        print(f"Edad: {self.edad}")
        print(f"Promedio: {self.promedio}")

if __name__ == "__main__":
    # Crear 3 instancias
    estudiante1 = Estudiante("Oscar", 20, 4.5)
    estudiante2 = Estudiante("Maria", 19, 4.8)
    estudiante3 = Estudiante("Juan", 21, 3.9)

    # Almacenarlas en una lista
    lista_estudiantes = [estudiante1, estudiante2, estudiante3]

    # Mostrar información de cada uno
    print("Información de los estudiantes en la lista:")
    for est in lista_estudiantes:
        est.mostrarInfo()
        print("-" * 20)

File: Clases_Instancias/2/estudiantes_lista.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Estudiante {
public:
    string nombre;
    int edad;
    float promedio;

    Estudiante(string n, int e, float p) : nombre(n), edad(e), promedio(p) {}

    void mostrarInfo() {
        cout << "Nombre: " << nombre << endl;
        cout << "Edad: " << edad << endl;
        cout << "Promedio: " << promedio << endl;
    }
};

int main() {
    // Definir un vector para almacenar las instancias
    vector<Estudiante> listaEstudiantes;

    // Crear 3 instancias y agregarlas al vector
    listaEstudiantes.push_back(Estudiante("Oscar", 20, 4.5));
    listaEstudiantes.push_back(Estudiante("Maria", 19, 4.8));
    listaEstudiantes.push_back(Estudiante("Juan", 21, 3.9));

    // Mostrar información de cada estudiante recorriendo el vector
    cout << "Informacion de los estudiantes en el vector:" << endl;
    for (int i = 0; i < listaEstudiantes.size(); i++) {
        listaEstudiantes[i].mostrarInfo();
        cout << "--------------------" << endl;
    }

    return 0;
}
